### PR TITLE
drop support for python27

### LINKED ISSUE #433: py.test for the default populus init project fails with python 2
```py.test test_greeter.py 
/usr/local/lib/python2.7/dist-packages/populus/__init__.py:13: DeprecationWarning: Python 2 support will end during the first quarter of 2018Please upgrade to Python 3https://medium.com/@pipermerriam/dropping-python-2-support-d781e7b48160
  warnings.warn(warn_msg, DeprecationWarning)
==================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.14, pytest-3.2.5, py-1.5.2, pluggy-0.4.0
rootdir: /home/solidity/testing/tests, inifile:
plugins: populus-2.2.0
collected 2 items                                                                                                                                                                             

test_greeter.py FF

========================================================================================== FAILURES ==========================================================================================
________________________________________________________________________________________ test_greeter ________________________________________________________________________________________

chain = <populus.chain.tester.TesterChain object at 0x7f458e5c3650>

    def test_greeter(chain):
>       greeter, _ = chain.provider.get_or_deploy_contract('Greeter')

test_greeter.py:2: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/populus/contracts/provider.py:143: in get_or_deploy_contract
    deploy_kwargs=deploy_kwargs,
/usr/local/lib/python2.7/dist-packages/populus/contracts/provider.py:111: in deploy_contract
    contract_data = self.get_contract_data(contract_identifier)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <populus.contracts.provider.Provider object at 0x7f458d973890>, contract_identifier = 'Greeter'

    def get_contract_data(self, contract_identifier):
        """
            Returns a dictionary containing the compiler output for the given
            contract identifier.
            """
        for backend in self.provider_backends.values():
            try:
                return backend.get_contract_data(contract_identifier)
            except UnknownContract:
                continue
        else:
            raise UnknownContract(
                "No contracts found for the contract identifier '{0}'".format(
>                   contract_identifier,
                )
            )
E           UnknownContract: No contracts found for the contract identifier 'Greeter'

/usr/local/lib/python2.7/dist-packages/populus/contracts/provider.py:171: UnknownContract
____________________________________________________________________________________ test_custom_greeting ____________________________________________________________________________________

chain = <populus.chain.tester.TesterChain object at 0x7f458d8ee590>

    def test_custom_greeting(chain):
>       greeter, _ = chain.provider.get_or_deploy_contract('Greeter')

test_greeter.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/populus/contracts/provider.py:143: in get_or_deploy_contract
    deploy_kwargs=deploy_kwargs,
/usr/local/lib/python2.7/dist-packages/populus/contracts/provider.py:111: in deploy_contract
    contract_data = self.get_contract_data(contract_identifier)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <populus.contracts.provider.Provider object at 0x7f458d87f950>, contract_identifier = 'Greeter'

    def get_contract_data(self, contract_identifier):
        """
            Returns a dictionary containing the compiler output for the given
            contract identifier.
            """
        for backend in self.provider_backends.values():
            try:
                return backend.get_contract_data(contract_identifier)
            except UnknownContract:
                continue
        else:
            raise UnknownContract(
                "No contracts found for the contract identifier '{0}'".format(
>                   contract_identifier,
                )
            )
E           UnknownContract: No contracts found for the contract identifier 'Greeter'

/usr/local/lib/python2.7/dist-packages/populus/contracts/provider.py:171: UnknownContract
====================================================================================== warnings summary ======================================================================================```

### PR DESCRIPTION
**Pull request by voith**
### What was wrong?

replaces #466. Some changes on top of work done by @cryptomental in #466. I have cherry-picked their commits. 
closes #469 
closes #433 
### How was it fixed?
- dropped support for `python2.7`
- dropped six module which was no longer needed.
- pinned python version in `setup.py`
- added new style super calls

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -46,8 +47,6 @@
         'Intended Audience :: Developers',
         'License :: OSI Approved :: BSD License',
         'Natural Language :: English',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.3',
```
Comment (veox) [inline setup.py]: Might want to remove this if `python_requires` is as above (`>=3.4,<4`).

**Comment by veox** [comment]
Might want to add to the PR description something like:

```
Closes #465 (done).
Closes #433 (obsolete).
```

**Comment by voith** [comment]
@veox @carver. Thank you so much for taking the time to review it!

**Comment by voith** [comment]
cc @miohtama. This PR has received two approvals and I've addressed the suggestion given!

**Pull request merged**